src/gst.c

```diff
@@ -0,0 +1,218 @@
+// GStreamer RTP streaming implementation
+
+#include <signal.h>
+#include <pthread.h>
+#include "cloudcam/cloudcam.h"
+#include "cloudcam/log.h"
+#include "cloudcam/gst.h"
+
+// creates a jpeg snapshot of the next available frame
+// buffer returned in *data must be released via free()
+size_t gst_get_jpeg_snapshot(gst_thread_ctx *ctx, void **data)
```

Comment:
this is probably a dumb question but why not just make it `*data`?

Reply:
gst_get_jpeg_snapshot() is returning a pointer here, not writing to a supplied buffer so it has to be void** (or void* return value with size returned via size_t* param)

Reply:
Made gst_get_jpeg_snapshot() to return the pointer instead with size passed via the size_t* argument